.github/workflows/lint-code.yml

```diff
@@ -56,3 +62,74 @@ jobs:
       - name: mypy --> Static type checking
         # Configured in pyprojet.toml
         run: mypy **/*.py
+
+  # Use pipreqs to check for missing dependencies
+  pipreqs-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install pipreqs
+        run: pip install pipreqs
+
+      - name: Install requirements
+        run: pip install -r requirements.txt
+
+      - name: Run pipreqs
+        run: pipreqs --savepath pipreqs.txt
+
+      - name: Compare requirements
+        run: |
+          # Extract and sort package names
+          awk '{print $1}' $1 | sort -u > "$1".compare
+          awk -F'==' '{print $1}' $2 | sort -u > "$2".compare
+
+          # Compare package lists
+          if cmp -s "$1".compare "$2".compare
+          then
+            echo "Requirements are the same"
+            exit 0
+          else
+            echo "Requirements are different"
+            exit 1
+          fi
+
+    # Use Prettier to check various file formats
+  prettier:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+
+      - name: Install Prettier
+        run: npm install -g prettier
+
+      - name: Run Prettier --check
+        run: prettier --check .
+
+  # Use editorconfig to check all remaining file formats
+  editorconfig:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+
+      - name: Install editorconfig-checker
+        run: npm install -g editorconfig-checker
+
+      - name: editorconfig --> Lint files
+        run: editorconfig-checker $(git ls-files | grep -v 'test\|.py\|.md\|.json\|.yml\|.yaml\|.html')
```

Comment:
This would exclude any file with `test` in it's name right? Is that intended?

Reply:
Yeah, now that you mention it I don't see a reason for it, so I'll exclude it in [b28b256](https://github.com/SciLifeLab/scilifelab_epps/pull/299/commits/b28b256de63be196daaac7a0e126a4433af5639c).